q2_boots/core_metrics.py

```diff
@@ -32,24 +33,71 @@ def core_metrics(ctx, table, sampling_depth, metadata,
 
     tables = bootstrapped_tables.values()
 
+    alpha_metric_names = alpha_metrics
+
+    alpha_metrics = {'PHYLO': {}, 'NONPHYLO': {}}
+
+    for metric in alpha_metric_names:
+        if metric in ALPHA_METRICS["PHYLO"]["IMPL"] or metric in\
+           ALPHA_METRICS["PHYLO"]["UNIMPL"]:
+            metric = ALPHA_METRICS['NAME_TRANSLATIONS'][metric]
+            alpha_metrics['PHYLO'][metric] = ctx.get_action('diversity_lib',
+                                                            metric)
+        else:
+            if metric in ALPHA_METRICS['NONPHYLO']['IMPL']:
+                metric = ALPHA_METRICS['NAME_TRANSLATIONS'][metric]
+                alpha_metrics['NONPHYLO'][metric] = ctx.get_action('diversity_lib',
+                                                                   metric)
+            else:
+                alpha_metrics['NONPHYLO'][metric] = ctx.get_action('diversity_lib',
+                                                                   'alpha_passthrough')
+
+    beta_metric_names = beta_metrics
+
+    beta_metrics = {'PHYLO': {}, 'NONPHYLO': {}}
+
+    for metric in beta_metric_names:
+        if metric in BETA_METRICS["PHYLO"]["IMPL"] or metric in\
+           BETA_METRICS["PHYLO"]["UNIMPL"]:
+            if metric in ('unweighted_unifrac', 'weighted_unifrac'):
+                metric = BETA_METRICS['NAME_TRANSLATIONS'][metric]
+                beta_metrics['PHYLO'][metric] = ctx.get_action('diversity_lib', metric)
+            else:
+                # handle unimplemented unifracs
```

Comment:
Same deal as alpha with the default case

Reply:
This one shouldn't change I don't think, it handles all unimplemented methods. I think this should be a default case because that makes sense. This is how diversity alpha and beta handle this as well.

Reply:
What's the deal with unimplemented methods here? Like what does that mean in this context? That sounds like they're asking for something we don't have which should be an error, but I'm assuming that's not quite what's happening.

Reply:
To be honest I don't know what it means, it runs a method called beta-passthrough in diversity-lib

Reply:
@gregcaporaso can you shed some light on this?

Reply:
Yep, the unifrac package provides a bunch of variants on weighted and unweighted unifrac, and we support some but not all of those. I don't recall exactly why it was implemented this way in q2-diversity-lib, but I do know that the variants that we do/don't support isn't as simple as a list of string identifiers for the metrics. In some cases two variants of a metric are toggled based on whether a parameter is passed to the underlying function in the unifrac packaged (e.g., `method_xa` and `method_xb` are both listed as `method_x`, and the function `method_x` takes a parameter that toggles between `a` and `b`). 

Do you guys need more info on this to move forward with this PR?